azure/scope/clients.go

```diff
@@ -169,3 +205,151 @@ func setValue(settings auth.EnvironmentSettings, key string) {
 		settings.Values[key] = v
 	}
 }
+
+func (c *AzureClients) reconcileASOSecret(ctx context.Context, asoClientSecret *ASOSecretClient) error {
+	newASOSecret := &corev1.Secret{
+		Data: map[string][]byte{
+			"AZURE_SUBSCRIPTION_ID": []byte(asoClientSecret.SubscriptionID),
+		},
+	}
+
+	var owner metav1.OwnerReference
+
+	if asoClientSecret.AzureClusterIdentity != nil {
+		newASOSecret.ObjectMeta = metav1.ObjectMeta{
+			Name:      aso.GetASOSecretName(asoClientSecret.AzureClusterIdentity.GetName(), aso.ClusterIdentityOwner, asoClientSecret.SubscriptionID),
+			Namespace: asoClientSecret.AzureClusterIdentity.GetNamespace(),
+		}
+
+		newASOSecret.Data["AZURE_TENANT_ID"] = []byte(asoClientSecret.AzureClusterIdentity.Spec.TenantID)
+		newASOSecret.Data["AZURE_CLIENT_ID"] = []byte(asoClientSecret.AzureClusterIdentity.Spec.ClientID)
+
+		apiVersion, kind := infrav1.GroupVersion.WithKind(asoClientSecret.AzureClusterIdentity.Kind).ToAPIVersionAndKind()
+		owner = metav1.OwnerReference{
+			APIVersion: apiVersion,
+			Kind:       kind,
+			Name:       asoClientSecret.AzureClusterIdentity.GetName(),
+			UID:        asoClientSecret.AzureClusterIdentity.GetUID(),
+		}
+
+		// Fetch identity secret, if it exists
+		key := types.NamespacedName{
+			Namespace: asoClientSecret.AzureClusterIdentity.Spec.ClientSecret.Namespace,
+			Name:      asoClientSecret.AzureClusterIdentity.Spec.ClientSecret.Name,
+		}
+		identitySecret := &corev1.Secret{}
+		err := asoClientSecret.KubeClient.Get(ctx, key, identitySecret)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to fetch identity secret %s as defined in AzureClusterIdentity %q/%q: %w", identitySecret.Name, asoClientSecret.AzureClusterIdentity.Namespace, asoClientSecret.AzureClusterIdentity.Name, err)
+		}
+
+		switch asoClientSecret.AzureClusterIdentity.Spec.Type {
+		case infrav1.ServicePrincipal, infrav1.ManualServicePrincipal:
+			newASOSecret.Data["AZURE_CLIENT_SECRET"] = identitySecret.Data[azureSecretKey]
+		case infrav1.ServicePrincipalCertificate:
+			newASOSecret.Data["AZURE_CLIENT_CERTIFICATE"] = identitySecret.Data["certificate"]
+			newASOSecret.Data["AZURE_CLIENT_CERTIFICATE_PASSWORD"] = identitySecret.Data["password"]
+		case infrav1.UserAssignedMSI:
+			newASOSecret.Data["AZURE_IDENTITY_RESOURCE_ID"] = []byte(asoClientSecret.AzureClusterIdentity.Spec.ResourceID)
```

Comment:
@adriananeci Could you share where you found to set `AZURE_IDENTITY_RESOURCE_ID` here? I can't find that string anywhere in the ASO docs or source code.

Reply:
You're right, there is no such thing. Managed identity is handled via Workload Identity or the deprecated aad-pod-identity, https://azure.github.io/azure-service-operator/guide/authentication/credential-format/#deprecated-managed-identity-aad-pod-identity (here it is mentioned about `IDENTITY_RESOURCE_ID`) .